app/src/main/java/com/example/cattlelog/MainActivity.kt

```diff
@@ -184,14 +194,44 @@ class MainActivity : AppCompatActivity() {
         return true
     }
 
-    private fun getDatabaseVersion() {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        Log.d("onActResult", "reqCode: $requestCode,  resultCode: $resultCode, data: $data")
+        if (requestCode == DB_DOWNLOAD_CODE){
+            if (resultCode == Activity.RESULT_OK){
+                getDatabaseVersion(true)
+            }
+        }
+    }
+
+    fun setDatabaseVersion(newestDBVersion: String){
+        val prefs = this.getSharedPreferences(PREFS_FILENAME, Context.MODE_PRIVATE)
+        val editor = prefs!!.edit()
+        editor.putString(DB_VERSION, newestDBVersion)
+        editor.apply()
+        val localVersion = prefs.getString(DB_VERSION, "NONE")
+        Log.d("dropboxAPI", "The stored database version has been set to $localVersion")
+    }
+
+    fun checkDatabaseCurrent(newestDBVersion: String): Boolean{
+        val prefs = this.getSharedPreferences(PREFS_FILENAME, Context.MODE_PRIVATE)
+        val localVersion = prefs!!.getString(DB_VERSION, "NONE")
+        if (localVersion == newestDBVersion){
+            return true
+        }
+        return false
+    }
+
+    // Sets database version after database is successfully downloaded (call it with getDatabaseVersion(true) for this functionality)
+    // Triggers new database download if local database is out of date (call it with getDatabaseVersion(false) for this functionality)
+    fun getDatabaseVersion(download_finished: Boolean) {
```

Comment:
This is confusing; we may need to reconsider our approach so it's easier to follow what this function does.

Roughly:

1. When `MainActivity` launches, get the latest database timestamp. Call this `getLatestDatabaseTimestamp`.

2. Compare this to the shared preference, if one exists. If either the preference doesn't exist, or the timestamp stored in it doesn't match the one from above, download the database and update the shared preference. Otherwise, do nothing—we have the latest database.

I suggest we create a single function called `syncWithLatestDatabase` that does both #1 and #2 above. Then, whenever we need to sync, simply call this function without any arguments.